system-variables.md

```diff
@@ -4284,6 +4284,107 @@ mysql> desc select count(distinct a) from test.t;
+- This addresses the same query patterns as variable [tidb_opt_ordering_index_selectivity_threshold](#tidb_opt_ordering_index_selectivity_threshold-new-in-v700).
+- It differs in implementation by applying a ratio or percentage of the possible range that the qualified rows will be found.
+- Value -1 (default) disables this ratio. Any value between 0 and 1 applies a ratio of 0% to 100% (meaning 0.5 = 50%).
+- In the following examples, table `t` has a total of 1,000,000 rows. The same query is used, but different values for `tidb_opt_ordering_index_selectivity_ratio` are used. The query in the example has a WHERE clause predicate that qualifies a small percentage of the rows (9000 out of 1,000,000). There is an index that supports the `ORDER BY a` (index ia), but the filtering on b does not appear in this index. Based upon the data distribution, the row matching the WHERE clause and LIMIT 1 could be found as the 1st row accessed when scanning the non-filtering index, or at worst, after nearly all of the rows have been processed.
+- The first example uses the default -1, which uses the existing estimation formula. The default behavior is that a small percentage of the rows are estimated to be scanned before a row is found that qualifies from filtering outside of that index.
+
+```sql
+> SET SESSION tidb_opt_ordering_index_selectivity_ratio = -1;
+
+>EXPLAIN SELECT * FROM t USE INDEX (ia) WHERE b <= 9000 ORDER BY a 1;
++-----------------------------------+---------+-----------+-----------------------+---------------------------------+
+| id                                | estRows | task      | access object         | operator info                   |
++-----------------------------------+---------+-----------+-----------------------+---------------------------------+
+| Limit_12                          | 1.00    | root      |                       | offset:0, count:1               |
+| └─Projection_22                   | 1.00    | root      |                       | test.t.a, test.t.b, test.t.c    |
+|   └─IndexLookUp_21                | 1.00    | root      |                       |                                 |
+|     ├─IndexFullScan_18(Build)     | 109.20  | cop[tikv] | table:t, index:ia(a)  | keep order:true                 |
+|     └─Selection_20(Probe)         | 1.00    | cop[tikv] |                       | le(test.t.b, 9000)              |
+|       └─TableRowIDScan_19         | 109.20  | cop[tikv] | table:t               | keep order:false                |
++-----------------------------------+---------+-----------+-----------------------+---------------------------------+
+```
+
+- The 2nd example uses 0, which assumes that 0% of the rows will be scanned before the qualified rows are found. 
+
+```sql
+>SET SESSION tidb_opt_ordering_index_selectivity_ratio = 0;
+
+>EXPLAIN SELECT * FROM t USE INDEX (ia) WHERE b <= 9000 ORDER BY a 1;
```

Comment:
I assume the reason we need this rectification is to make better decision of table/index access. 
In the examples, forcing index access doesn't reflect the decision change caused by the variable.  Is it a better idea if we remove the hints?  The different values switch table/index access.

Reply:
The hints are there to demonstrate the effect on the estimated rows. Whether this new estimate results in that index losing to another plan is entirely dependent on the cost of other plans.